Add description fields to all Hyprland key bindings.

ashebanow opened the pull request:
# Pull Request

## Description

This pull request adds description fields to all Hyprland key bindings. This makes the file more readable and easier to understand for new users. But just as importantly, the descriptions will make it possible to use a which-key style help system like the example in [the HyprHook plugin](https://github.com/Hyprhook/Hyprhook/tree/master).

## Type of change

Please put an `x` in the boxes that apply:

- [ ] **Bug fix** (non-breaking change which fixes an issue)
- [x] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that would cause existing functionality to not work as expected)
- [ ] **Documentation update** (non-breaking change; modified files are limited to the documentations)
- [ ] **Technical debt** (a code change that does not fix a bug or add a feature but makes something clearer for devs)
- [ ] **Other** (provide details below)

## Checklist

Please put an `x` in the boxes that apply:

- [x] I have read the [CONTRIBUTING](https://github.com/JaKooLit/Hyprland-Dots/blob/main/CONTRIBUTING.md) document.
- [x] My code follows the code style of this project.
- [x] My commit message follows the [commit guidelines](https://github.com/JaKooLit/Hyprland-Dots/blob/main/CONTRIBUTING.md#git-commit-messages).
- [ ] My change requires a change to the documentation.
- [ ] I want to add something in Hyprland-Dots wiki.
- [ ] I have added tests to cover my changes. [NOTE: I didn't find any tests to extend]
- [x] I have tested my code locally and it works as expected.
- [ ] All new and existing tests passed. [NOTE: I didn't find any tests to run]

ashebanow commented on config/hypr/UserConfigs/UserKeybinds.conf:
```diff
@@ -13,27 +13,27 @@ $UserScripts = $HOME/.config/hypr/UserScripts
-bind = $mainMod SHIFT, O, exec, $UserScripts/ZshChangeTheme.sh # Change oh-my-zsh theme
-bindn = ALT_L, SHIFT_L, exec, $scriptsDir/SwitchKeyboardLayout.sh # Changing the keyboard layout
+bind = $mainMod SHIFT, O, "Change ZSH Theme", exec, $UserScripts/ZshChangeTheme.sh # Change oh-my-zsh theme
+bindn = ALT_L, SHIFT_L, "Switch Keybolard Layout", exec, $scriptsDir/SwitchKeyboardLayout.sh # Changing the keyboard layout
```
Typo in "Keybolard" - how did I miss that?!?!

ashebanow replied:
Fixed

JaKooLit commented:
Thanks for this.

however, most of the explanations are already at the end.. See example below

bind = $mainMod, Z, "Toogle Screen Zoom", exec, pypr zoom **# Toggle Zoom**

ashebanow commented:
For sure, and in fact I used the comment text as a starting point for the
description fields.

But it's a lot more useful to have the descriptions in a form that can be
accessed programmatically.

This PR is just part 1. Second step is a cheatsheet panel in ags that
leverages that data.

On Tue, Jul 30, 2024 at 8:12 PM Ja.KooLit ***@***.***> wrote:

> Thanks for this.
>
> however, most of the explanations are already at the end.. See example
> below
>
> bind = $mainMod, Z, "Toogle Screen Zoom", exec, pypr zoom *# Toggle Zoom*
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/JaKooLit/Hyprland-Dots/pull/390#issuecomment-2259555589>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAACX7ZSQ2XZ7AKZHARWEW3ZPBI3LAVCNFSM6AAAAABLXMPYASVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDENJZGU2TKNJYHE>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>

JaKooLit commented:
a great.. understood..

I will review first on my system later. atm just using mobile browser

JaKooLit commented:
![Screenshot_31-Jul_18-25-25_5713](https://github.com/user-attachments/assets/c4d06c8a-6a8f-44fb-806e-5f1b486eb980)

so... I have tried....

The moment I changed it, got so much errors

I am using the hyprland-git version in case you are wondering

ashebanow commented:
Was out this morning, only just now got time to take a look. Something very
weird is going on. I wasn't seeing the problem, but when I updated my
EndeavourOS via pacman -Syu, it broke. Will dig further, ignore for now.

On Wed, Jul 31, 2024 at 2:27 AM Ja.KooLit ***@***.***> wrote:

> Screenshot_31-Jul_18-25-25_5713.png (view on web)
> <https://github.com/user-attachments/assets/c4d06c8a-6a8f-44fb-806e-5f1b486eb980>
>
> so... I have tried....
>
> The moment I changed it, got so much errors
>
> I am using the hyprland-git version in case you are wondering
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/JaKooLit/Hyprland-Dots/pull/390#issuecomment-2260073758>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAACX754X75F7KE6T5A34MDZPCUZXAVCNFSM6AAAAABLXMPYASVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDENRQGA3TGNZVHA>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>
